### DISCUSSION TITLE
Email configuration, admin and self hosted questions

### DISCUSSION THREAD

**Discussion by jxs714** [documenso/documenso#695] [General]
Hey guys,

A few questions to ask about the software:

- Is the self hosted option limited to only 5 documents per month as the free version on your website? Or is it unlimited? I see a limit in the main dashboard prior to confirming the email address (Which i cant since I dont have the smtp config done properly)
- Can we use our own server smtp with postfix to connect documenso for emails by using login credentials similar to any email software? If so, where is the configuration settings? Or is inbucket required in order for emails to work? 
- Is there an admin UI to customize the software?

Please advise, thanks.

**Comment by gabriel-blankenship** [comment]
I have been trying to figure out a similar configuration with postfix, they posted this was supposedly possible via Discord but no details as to how specifically within the .env file.  The [[SMTP]] section of the .env isn't exactly clear on how to get that specific configuration working.

**Comment by Mythie** [comment]
Hey there, based on how others have done Nodemailer with postfix you should be able to get away with something similar to the following:

```
# [[SMTP]]
# OPTIONAL: Defines the transport to use for sending emails. Available options: smtp-auth (default) | smtp-api | mailchannels
NEXT_PRIVATE_SMTP_TRANSPORT="smtp-auth"
# OPTIONAL: Defines the host to use for sending emails.
NEXT_PRIVATE_SMTP_HOST="127.0.0.1"
# OPTIONAL: Defines the port to use for sending emails.
NEXT_PRIVATE_SMTP_PORT=25 # Postfix port
# OPTIONAL: Defines the username to use with the SMTP server.
NEXT_PRIVATE_SMTP_USERNAME=""
# OPTIONAL: Defines the password to use with the SMTP server.
NEXT_PRIVATE_SMTP_PASSWORD=""
```

I did note that others have to disable the rejecting of unauthorised TLS connections so that may be an issue that has to be solved for.

As an aside do you need to run Postfix? I assume you're using it as a relay or bridge to a different SMTP server which could instead be configured above.

**Comment by gabriel-blankenship** [comment]
for a locally configured relay I'm currently seeing this in the logs due to a self signed certificate the application does not seem to want to accept.

```
Feb 27 13:14:04 esign postfix/smtpd[13121]: connect from localhost[127.0.0.1]
Feb 27 13:14:04 esign postfix/smtpd[13121]: SSL_accept error from localhost[127.0.0.1]: lost connection
Feb 27 13:14:04 esign postfix/smtpd[13121]: lost connection after STARTTLS from localhost[127.0.0.1]
Feb 27 13:14:04 esign postfix/smtpd[13121]: disconnect from localhost[127.0.0.1] ehlo=1 starttls=0/1 commands=1/2
```

Would adding a value in the following resolve this?

```
# OPTIONAL: Defines whether to force the use of TLS.
NEXT_PRIVATE_SMTP_SECURE=
```

And if so what might that value be?  Treating it like a boolean did nothing.

**Reply by Mythie** [reply]
Secure is by default set to false however there will need to be an additional item for ignore unauthorized connections as per the above.

This additional item is a new change that would need to be done in order to fully support postfix.

**Comment by gabriel-blankenship** [comment]
@Mythie 
went ahead and setup sendmail as well to see if that would make any difference and am not getting any further really, though I did receive a different message.

```
Feb 27 16:04:01 esign sm-mta[16242]: 41RG416g016242: localhost [127.0.0.1] did not issue MAIL/EXPN/VRFY/ETRN during connection to MTA-v4
```

Is this a change that will need to be made to the web application or is this something we as self-hosting users will need to attempt to resolve?  Additional logging may help with that, but not sure how that may be accomplished at present.

**Comment by compgeniuses** [comment]
what settings have worked for all of you, i seem to be getting issues sending the confirmation message

**Reply by lookingforthedoor** [reply]
Did you ever get a response?  We are running into the same issue.

**Reply by compgeniuses** [reply]
We are impelmenting Documenso for Nethserver 8 here https://github.com/geniusdynamics/ns8-documenso

Everything works fine, but emails seems not to be working. i have never received a resposne as well

**Comment by astromatiq** [comment]
For me, **port 587** worked with **NEXT_PRIVATE_SMTP_SECURE=false.** Although port 465 is normally working as intended on my mail hosting side.
